### PR TITLE
Test all the things

### PR DESCRIPTION
**Pull request by TiddoLangerak**
Additionally:
- Automatic builds on travis-ci
- Build status icon in the README
- Some fixes in error handling ( @alexnederlof you were right, should've tested this in the first place already)

@Magnetme/monolith (especially @wouter-willems) RFR!

### CODE REVIEW

File: test/mock.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+//setter on the window object that will copy angular to the global object
+Object.defineProperty(window, 'angular', {
+	set(val) {
+		global.angular = val;
+	},
+	get() {
+		return global.angular;
+	}
+
+});
```
Comment (TiddoLangerak) [inline test/mock.js]: I'm going to move this file to a standalone project if we're going to use this more often, but for now I figured it was not worth the effort

File: src/index.js [hunk 2]
```diff
@@ -5,14 +5,29 @@ export default angular.module('mm.$async', [])
...
+							return;
+						}
+
 						if (state.done) {
 							resolve(state.value);
 						} else {
 							$q.when(state.value)
-								.then(next, err => it.throw(err));
+								.then(next, err => {
+									next(err, true);
```
Comment (wouter-willems) [inline src/index.js]: why is this changed from throwing an error?

**Comment by wouter-willems** [comment]
great work :+1:

File: src/index.js (hunk 2 above)
Comment (TiddoLangerak) [inline src/index.js]: When an error is triggered then the generator is still able to catch the exception. However, in the previous implementation I would throw and then quit. This means that the generator could catch the error, but would get stuck on the next `yield`. 

With this change we still throw the error though, but now in the next function on line 18. This allows the engine to properly resume after throwing the error.

This behaviour is by the way described in [this test case](https://github.com/Magnetme/ng-async/pull/2/files#diff-c1129c8b045390789fa8ff62f2c6b4a9R101). The previous implementation did not pass there, the current does.

Comment (wouter-willems) [inline src/index.js]: great explanation, thnx
